### PR TITLE
Merge networks api

### PR DESCRIPTION
**Pull request by obrix**
- [x ] The commit message follows our guidelines
- [x ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

This is the implementation for issue https://github.com/powsybl/pypowsybl/issues/212 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

No api to merge networks was available.

**What is the new behavior (if this is a feature change)?**

An api to merge networks is available, this is the core api wrapped and working the same way.

### CODE REVIEW

**Comment by obrix** [comment]
I used a specific C struct to pass an array of ObjectHandles from C++ to java. As discussed with @geofjamg there may be a more direct way to do this by defining and using a Java structure extending PointerBase with  @CPointerTo(VoidPointer.class) annotation. I will test that soon to see if this works. The solution provided here is operational though.

File: java/src/main/java/com/powsybl/python/PyPowsyblApiHeader.java [hunk 1]
```diff
@@ -52,6 +57,17 @@ private PyPowsyblApiHeader() {
         void setLength(int length);
     }
 
+    /* Import of a C function pointer type. */
+    interface GetElementFunctionPointer extends CFunctionPointer {
```
Comment (geofjamg) [inline java/src/main/java/com/powsybl/python/PyPowsyblApiHeader.java]: We can remove it?

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: tests/test_network.py [hunk 2]
```diff
@@ -524,6 +524,21 @@ def test_create_network(self):
         n = pp.network.create_ieee118()
         self.assertEqual('ieee118cdf', n.id)
 
+    def test_network_merge(self):
+        n1 = pp.network.load(str(TEST_DIR.joinpath('empty-network.xml')))
```
Comment (geofjamg) [inline tests/test_network.py]: We have to find a better test case, with mergeable networks.

File: java/src/main/java/com/powsybl/python/PyPowsyblApiHeader.java (hunk 1 above)
Comment (obrix) [reply java/src/main/java/com/powsybl/python/PyPowsyblApiHeader.java]: Yes, I removed it.

**Review by geofjamg — APPROVED** [review]

**Pull request merged**